services/submission.js

```diff
@@ -0,0 +1,129 @@
+const { NEW, IN_PROGRESS, SUBMITTED, RELEASED, COMPLETED, ARCHIVED} = require("../constants/submission-constants");
+const {v4} = require('uuid')
+const {getCurrentTime} = require("../crdc-datahub-database-drivers/utility/time-utility");
+const {HistoryEventBuilder} = require("../domain/history-event");
+const {verifySession} = require("../verifier/user-info-verifier");
+const {getSortDirection} = require("../crdc-datahub-database-drivers/utility/mongodb-utility");
+const {formatName} = require("../utility/format-name");
+const ERROR = require("../constants/error-constants");
+const USER_CONSTANTS = require("../crdc-datahub-database-drivers/constants/user-constants");
+const ROLES = USER_CONSTANTS.USER.ROLES;
+const ALL_FILTER = "All";
+
+
+
+function listConditions(userID, userRole, aUserOrganization, params){
+    const validApplicationStatus = {status: {$in: [NEW, IN_PROGRESS, SUBMITTED, RELEASED, COMPLETED, ARCHIVED]}};
+    // Default conditons are:
+    // Make sure application has valid status
+    let conditions = {...validApplicationStatus};
+    // Filter on organization and status
+    if (params.organization !== ALL_FILTER) {
+        conditions = {...conditions, "organization.name": params.organization};
+    }
+    if (params.status !== ALL_FILTER) {
+        conditions = {...conditions, status: params.status};
+    }
+    // List all applications if Fed Lead / Admin / Data Concierge / Data Curator
+    const listAllApplicationRoles = [ROLES.ADMIN, ROLES.FEDERAL_LEAD, ROLES.CURATOR, ROLES.DC_POC];
+    if (listAllApplicationRoles.includes(userRole)) {
+        return [{"$match": conditions}];
+    }
+    // If org owner, add condition to return all data submissions associated with their organization
+    if (userRole === ROLES.ORG_OWNER && aUserOrganization?.orgName) {
+        conditions = {...conditions, "organization.name": aUserOrganization.orgName};
+        return [{"$match": conditions}];
```

Comment:
Could you double check the requirement? Currently, if you belong to an organization, you wont be able to fetch any submission submitted by submitterID.

Reply:
Requirements say that org owners don't need to filter by submitterID